cmd/lvldb2elastic/databasereader/databaseReader.go

```diff
@@ -0,0 +1,251 @@
+package databasereader
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/factory"
+	"github.com/ElrondNetwork/elrond-go/storage/leveldb"
+	leveldb2 "github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+var log = logger.GetOrCreate("databasereader")
+
+const shardBlocksStorer = "BlockHeaders"
+const metaHeadersStorer = "MetaBlock"
+const miniBlockStorer = "MiniBlocks"
+const maxOpenFiles = 100
+
+type DatabaseInfo struct {
+	Epoch uint32
+	Shard uint32
+}
+
+type databaseReader struct {
```

Comment:
this should be refactored in a way that can be easily tested. Remember the lessons learned from pruningStorer.